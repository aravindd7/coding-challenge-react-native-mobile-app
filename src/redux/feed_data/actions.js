export const FEED_DATA__UPDATE = "FEED_DATA__UPDATE";

export const feedData_Update = ({ data }) => ({
  type: FEED_DATA__UPDATE,
  payload: { data }
});
