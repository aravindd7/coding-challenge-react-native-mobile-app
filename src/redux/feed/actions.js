import { feedData_Update } from "../feed_data/actions";

export const FEED_UPDATE__REQUEST = "FEED_UPDATE__REQUEST";
export const FEED_UPDATE__FAIL = "FEED_UPDATE__FAIL";
export const FEED_UPDATE__SUCCESS = "FEED_UPDATE__SUCCESS";

export const FEED_SECTION_CHANGE = "FEED_SECTION_CHANGE";

export const changeSection = ({ section }) => {
  return (dispatch, getState) => {
    // TODO changeSection Action Creator
  };
};

export const feedUpdate = () => {
  return (dispatch, getState) => {
    // TODO feedUpdate Action Creator
  };
};

export const feedUpdate_Request = () => ({
  type: FEED_UPDATE__REQUEST,
  payload: {}
});

export const feedUpdate_Fail = ({ error }) => ({
  type: FEED_UPDATE__FAIL,
  payload: { error }
});

export const feedUpdate_Success = () => ({
  type: FEED_UPDATE__SUCCESS,
  payload: {}
});

export const feedSectionChange = ({ section }) => ({
  type: FEED_SECTION_CHANGE,
  payload: { section }
});
