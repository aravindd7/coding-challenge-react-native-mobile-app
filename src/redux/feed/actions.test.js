import * as actions from "./actions";

it("Creates Action changeSection", () => {
  const actionCreator = actions.changeSection({ section: "top" });
  const dispatchMock = jest.fn();

  const apiCallPromise = actionCreator(dispatchMock);

  expect(dispatchMock.mock.calls).toMatchSnapshot();
});

it("Creates Action feedUpdate", async () => {
  const response = {
    data: {
      children: [
        {
          data: {
            id: "1234",
            title: "title",
            author: "author",
            num_comments: 3,
            score: 2,
            created_utc: 1517115254,
            thumbnail: "thumbnail.jpg",
            url: "http://example.com"
          }
        },
        {
          data: {
            id: "1234",
            title: "title",
            author: "author",
            num_comments: 3,
            score: 2,
            created_utc: 1517115254,
            thumbnail: "",
            url: "http://example.com",
            media: { oembed: { thumbnail_url: "thumbnail_url.jpg" } }
          }
        }
      ]
    }
  };

  window.fetch = jest
    .fn()
    .mockImplementation(() => Promise.resolve({ json: () => response }));

  const actionCreator = actions.feedUpdate();
  const dispatchMock = jest.fn();

  const apiCallPromise = actionCreator(dispatchMock, () => ({
    feed: { section: "top" }
  }));
  await apiCallPromise;

  expect(dispatchMock.mock.calls).toMatchSnapshot();
});

it("Creates Action feedUpdate with error", async () => {
  const response = {
    data: {
      children2: []
    }
  };

  window.fetch = jest
    .fn()
    .mockImplementation(() => Promise.resolve({ json: () => response }));

  const actionCreator = actions.feedUpdate();
  const dispatchMock = jest.fn();

  const apiCallPromise = actionCreator(dispatchMock, () => ({
    feed: { section: "top" }
  }));
  await apiCallPromise;

  expect(dispatchMock.mock.calls).toMatchSnapshot();
});

it("Creates Action feedUpdate_Request", () => {
  const action = actions.feedUpdate_Request();
  expect(action).toMatchSnapshot();
});

it("Creates Action feedUpdate_Fail", () => {
  const action = actions.feedUpdate_Fail({ error: "error message" });
  expect(action).toMatchSnapshot();
});

it("Creates Action feedUpdate_Success", () => {
  const action = actions.feedUpdate_Success();
  expect(action).toMatchSnapshot();
});

it("Creates Action feedSectionChange", () => {
  const action = actions.feedSectionChange({ section: "top" });
  expect(action).toMatchSnapshot();
});
