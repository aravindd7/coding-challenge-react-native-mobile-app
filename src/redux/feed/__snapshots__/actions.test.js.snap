// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates Action changeSection 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "section": "top",
      },
      "type": "FEED_SECTION_CHANGE",
    },
  ],
  Array [
    [Function],
  ],
]
`;

exports[`Creates Action feedSectionChange 1`] = `
Object {
  "payload": Object {
    "section": "top",
  },
  "type": "FEED_SECTION_CHANGE",
}
`;

exports[`Creates Action feedUpdate 1`] = `
Array [
  Array [
    Object {
      "payload": Object {},
      "type": "FEED_UPDATE__REQUEST",
    },
  ],
  Array [
    Object {
      "payload": Object {},
      "type": "FEED_UPDATE__SUCCESS",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "data": Array [
          Object {
            "author": "author",
            "created_utc": 1517115254,
            "date": 1517115254060,
            "id": "1234",
            "num_comments": 3,
            "score": 2,
            "thumbnail": "thumbnail.jpg",
            "title": "title",
            "url": "http://example.com",
          },
          Object {
            "author": "author",
            "created_utc": 1517115254,
            "date": 1517115254060,
            "id": "1234",
            "num_comments": 3,
            "score": 2,
            "thumbnail": "thumbnail_url.jpg",
            "title": "title",
            "url": "http://example.com",
          },
        ],
      },
      "type": "FEED_DATA__UPDATE",
    },
  ],
]
`;

exports[`Creates Action feedUpdate with error 1`] = `
Array [
  Array [
    Object {
      "payload": Object {},
      "type": "FEED_UPDATE__REQUEST",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "error": "Unable to fetch the feed",
      },
      "type": "FEED_UPDATE__FAIL",
    },
  ],
]
`;

exports[`Creates Action feedUpdate_Fail 1`] = `
Object {
  "payload": Object {
    "error": "error message",
  },
  "type": "FEED_UPDATE__FAIL",
}
`;

exports[`Creates Action feedUpdate_Request 1`] = `
Object {
  "payload": Object {},
  "type": "FEED_UPDATE__REQUEST",
}
`;

exports[`Creates Action feedUpdate_Success 1`] = `
Object {
  "payload": Object {},
  "type": "FEED_UPDATE__SUCCESS",
}
`;
